platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/TaskSchedulerBuilder.java

```diff
@@ -0,0 +1,346 @@
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.wiring;
+
+import static com.swirlds.logging.LogMarker.EXCEPTION;
+
+import com.swirlds.common.metrics.extensions.FractionalTimer;
+import com.swirlds.common.metrics.extensions.NoOpFractionalTimer;
+import com.swirlds.common.wiring.counters.BackpressureObjectCounter;
+import com.swirlds.common.wiring.counters.MultiObjectCounter;
+import com.swirlds.common.wiring.counters.NoOpObjectCounter;
+import com.swirlds.common.wiring.counters.ObjectCounter;
+import com.swirlds.common.wiring.counters.StandardObjectCounter;
+import com.swirlds.common.wiring.schedulers.ConcurrentTaskScheduler;
+import com.swirlds.common.wiring.schedulers.SequentialTaskScheduler;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.lang.Thread.UncaughtExceptionHandler;
+import java.time.Duration;
+import java.util.Objects;
+import java.util.concurrent.ForkJoinPool;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * A builder for {@link TaskScheduler}s.
+ *
+ * @param <O> the output type of the primary output wire for this task scheduler (use {@link Void} for a scheduler with
+ *            no output)
+ */
+public class TaskSchedulerBuilder<O> {
+
+    private static final Logger logger = LogManager.getLogger(TaskSchedulerBuilder.class);
+
+    public static final long UNLIMITED_CAPACITY = -1;
+
+    private final WiringModel model;
+
+    private boolean concurrent = false;
+    private final String name;
+    private TaskSchedulerMetricsBuilder metricsBuilder;
+    private long unhandledTaskCapacity = UNLIMITED_CAPACITY;
+    private boolean flushingEnabled = false;
+    private boolean externalBackPressure = false;
+    private ObjectCounter onRamp;
+    private ObjectCounter offRamp;
+    private ForkJoinPool pool = ForkJoinPool.commonPool();
+    private UncaughtExceptionHandler uncaughtExceptionHandler;
+
+    private Duration sleepDuration = Duration.ofNanos(100);
+
+    /**
+     * Constructor.
+     *
+     * @param name the name of the task scheduler. Used for metrics and debugging. Must be unique. Must only contain
+     *             alphanumeric characters and underscores.
+     */
+    TaskSchedulerBuilder(@NonNull final WiringModel model, @NonNull final String name) {
+        this.model = Objects.requireNonNull(model);
+
```

Comment:
Need a non-null check here

Reply:
I dereference `name` via `name.matches("^[a-zA-Z0-9_]*$")`, so if you pass in null it will throw an informative error before we update any data structures.